shared/record_linkage_shared/accept.py

```diff
@@ -66,14 +67,15 @@ def accept_matches(df_match, passnum, config):
 
     # create filters for high-similarity common_id so that it will skip
     # if the match does not compare common_id similarities
-    if 'common_id' in df_match.columns:
-        masks['common_id_null'] = df_match.common_id == -1
-        masks['id_high_mask'] = df_match.common_id >= thresholds['id_high_score']
-        masks['id_review_mask'] = df_match.common_id >= id_review_score
-    else:
-        masks['common_id_null'] = True
-        masks['id_high_mask'] = False
-        masks['id_review_mask'] = False
+    common_id_fields = [x for x in df_match.columns if "common_id" in x]
+    masks['common_id_null'] = len(common_id_fields) == 0
+    masks['id_high_mask'] = False
+    masks['id_review_mask'] = False
+
+    for common_id in common_id_fields:
+        masks['common_id_null'] = masks['common_id_null'] | (df_match[common_id] == -1)
```

Comment:
Doesn't this mean masks['common_id_null'] = True if one of the common ID is null? Am I misunderstanding?

Say we have common_id_ssn and common_id_id. common_id_ssn is null and common_id_id is not null.

masks['common_id_null'] starts as False.
Then, when common_id == 'common_id_ssn',  masks['common_id_null'] = False | True = True
Then, when common_id == 'common_id_id', masks['common_id_null'] = True | False = True
masks['common_id_null'] ends as True.

Reply:
You are right in terms of the logic but I think I might have done that on purpose. I made this change almost a month ago, so my memory is a little hazy, but I think my reasoning was if none of the IDs pass the ID high mask, we would want to give any case of null IDs the benefit of the doubt. I have this table drawn in my notes:

| ID 1/ ID 2 | Match | Don't Match | Null    |
| -------- | ------- | ------- | ------- |
| Match  | high mask   | high mask  | high mask |
| Don't Match | high mask   | no mask     | ? |
| Null    | high mask   | ? | null mask     |

And I believe I decided the ? marks should get the null mask instead of no mask. What do you think?

Reply:
I see your point and I can see it go either way. Basically we have half conflicting information, half uncertainty for these cases, and it kind of depends on the context whether we weigh conflicts more or uncertainty more. I think I'm okay with going with this current approach for now and see whether it leads to false matches during future testing.

Could you add a comment for the line where you define common_id_null mask that this definition considers common_id_null = True if at least one common_id is null? Just in case we want to tweak in the future.

I would also document this table somewhere...but I'm not sure what's the best way to document the multiple ID case yet. If you can just save it in your personal notes for now, we can discuss what to do for documentation after merging this PR.

Reply:
Yes, I'll add a note. It gets trickier if you start thinking about it in any dimension greater than 2 (like if 2 don't match and 1 is null, versus if 2 are null and 1 doesn't match). I wish we had nice data to test on, maybe we put an issue to check on the affect of this decision and best option to remind us when we do come across a good data set with multiple ids.

Reply:
That's a good idea! Could you add the issue?